### ISSUE TITLE
AirPlay playback partially not working

### ISSUE THREAD

**Issue by lc-jrx** [music-assistant/hass-music-assistant#2569]
### What version of Music Assistant has the issue?

2.1.0b10

### What version of the Home Assistant Integration have you got installed?

 2024.6.2

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

In my network there are six AirPlay devices. Unfortunately two of them refuse to play together with Music Assistant. With the other four devices there's no problem. The play my radio streams as expected.

Working AirPlay devices:
- 1x AirPort Express (latest version) - wireless
- 3x AirPort Express (latest version) - wired

Non-working AirPlay devices:
- 1x AirPort Express (latest version) - wireless - (name: kitchenspeaker)
- 1x HomePod Mini (latest version) - wireless - (name: badezimmer)

Whenever I start one of my two chosen radio streams on one of those two devices regarding to the log (debug) the streams starts, plays for two seconds and then aborts the stream with ffmpeg returncode "224", "-2", or "255". It doesn't matter if the devices were selected as solo speakers or as connected speakers grouped with on of the working devices.

Both devices just working fine with other sources. And also the other four devices just working fine with MA over AirPlay (both solo or grouped).

(b.tw. What a great software! Thank you very much!)

### How to reproduce

Choose one of the devices in target selection, press play on the selected radio station.

### Music Providers

Music Assistant, own streaming url

### Player Providers

AirPlay

### Full log output

[musicassistant.log](https://github.com/user-attachments/files/16110967/musicassistant.log)

### Additional information

tried various combinations of devices, tried various settings in the airplay device section, tried Apple Music as source

### What version of Home Assistant Core are your running

2024.6.4

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by marcelveldt** [comment]
Does it actually play that 2 seconds (so do you hear music) ?
Because most often of playback doesn't start and fails this way, it means that the communication with the player failed.
Note that there needs to be a 2 way communication possible between MA and the airplay player. 

So make sure the players are on the same subnet, no weird network filtering in between etc.

**Comment by eamonnsullivan** [comment]
I'm seeing issues too, on b9. Going to try and upgrade to b10. Music assistant is running in a docker container on Raspberry Pi, with Home Assistant 2024.7.1. My two Google speakers work (Chromecast), but not my two WiiM Minis.

```
2024-07-05 16:56:25.396 ERROR (MainThread) [music_assistant.webserver] Error handling message: player_queues/play_media: unhandled errors in a TaskGroup (1 sub-exception)
```

**Comment by eamonnsullivan** [comment]
Not sure if it was the restart, but b10 is giving me issues too. The music played for a few seconds and then stopped. Music Assistant appears to think there is music playing (displaying track info), but nothing is coming out of the speakers. 

[musicassistant2.log](https://github.com/user-attachments/files/16112294/musicassistant2.log)

**Comment by eamonnsullivan** [comment]
I've gone back to the "latest" tag (2.0.7) and still doesn't work, so I must not have the same issue. I tried rebooting the router. The Chromecast speakers still work, but not the Airplay ones. :man_shrugging:

**Comment by OzGav** [comment]
Per Marcels comment please advise if you have _anything_ unusual about your network setup.

**Comment by eamonnsullivan** [comment]
My network details: Asus ZenWifi AX router, with the main router downstairs and a mesh one upstairs. Both the airplay devices are downstairs. My router is set up to use one tri-band "smart connect" (2.4GHz and two 5GHz channels) under one SSID, which everything (except for the wired things) uses. The Pi with Home Assistant and Music Assistant on it (docker compose) is wired to the main router.

My airplay devices are two WiiM minis plugged into (via the aux connection) two Bose Soundtouch 10s. Airplay works from, say, my phone or via the WiiM app. Music Assistant clearly thinks it is sending music. There are no errors in the logs, and the UI is showing the track being played. 

I'm also using the SoundTouch Plus integration from HACS, but that *seems* to be working (speakers turn off/on, it shows the speakers as "playing", etc.)

I've tried multiple versions of Music Assistant (b9, b10 and 2.0.7) and the issue is the same, so I suspect something else went wrong. Unfortunately, I can't say when it stopped working because I've been on vacation for the last couple of weeks. I was playing music on this set up while packing, and when I came back and updated everything (Home Assistant to 2024.7.0, Music Assistant to b9 and the SoundTouch Plus integration to v1.0.55), it all stopped working.

**Comment by lc-jrx** [comment]
> Does it actually play that 2 seconds (so do you hear music) ? Because most often of playback doesn't start and fails this way, it means that the communication with the player failed.

No, there's nothing to hear. 2s are shown in the debug log. 2s for the AirPort Express, 0s for the HomePod. (debug log excerpts attached)
[debug_mass_airplay.txt](https://github.com/user-attachments/files/16115369/debug_mass_airplay.txt)


> Note that there needs to be a 2 way communication possible between MA and the airplay player.

That's clear. Thx. That was my mistake in the first place, but that's now fixed.

> So make sure the players are on the same subnet, no weird network filtering in between etc.

In my view that's ok and working. all AirPort Express devices are in the same vlan. Comm between those devices and the MA are granted. The HomePod works in another vlan and is reachable from the raspi.

**Comment by OzGav** [comment]
Are you are saying the devices are on different VLANs but you are trying to forward the communication between them?

**Comment by eamonnsullivan** [comment]
Interesting. Mine were set to "Everyone", and it started working when I switched the setting to "same network", it started working again. Thanks!

**Comment by vistalba** [comment]
> Eureka! I found the mistake. The two (in real three) non working devices are managed by HomeKit (don't know why ... all AirPlay devices are HomeKit joined). And HomeKit itself prevented the devices from playing. In the preferences section of HomeKit (iOS) there's a setting for AirPlay ("Speaker & TV"). This was switched to "Only persons in this home". After switching to "All in the same network" everything is working as expected.

That did the trick for me too! Many Thanks!

**Comment by altserg** [comment]
I have just set up HomePod mini, and have the same issue. Whatever settings g in homekit I try, the music is not played, and I cannot even control the volume , etc.
I use latest mass on too but not from container but directly install it using pip. I also had to patch magic raop binary to remove reference to libc.musl. I also tried to recompile raop on my system and substitute it in mass but the behavior is the same

Any ideas what is correct setup for homepod mini?

**Comment by OzGav** [comment]
@altserg you are using a non standard install method so we cannot help with that (and you are posting in a closed issue)
